### PR TITLE
Solved issue #62645 CI build fails on install pip packages 

### PR DESCRIPTION
**Pull request by giuliocn**
Here is a technical summary #62645 :
1. Use a script to build and install python from source 
2. Edit `Dockerfile.cpu` to execute `/install/build_and_install_python.sh 3.9.18` 
3. Edit `install_deb_packages.sh` to update and install packages
    - clang 
    - gcc
    - libffi-dev liblapack-dev libblas-dev
    - make
    - python3-numpy
4. Edit `install_pip_packages.sh` to install pip
    - 21, use python3.9 
    - 25, upgrade pip with python3.9
5. Edit `install_pip_packages.sh` to upgrade pip packages 
    - setuptools
    - future
    - numpy scipy joblib threadpoolctl scikit-learn pandas
    
IMPORTANT NOTES:
Running all tests with `bazel test` fails on **win_1803/py38** because of its `interpreter_path`
<details>
Running 'bazel test //tensorflow/...' inside tf_ci.cpu...
Reading package lists...
Building dependency tree...
Reading state information...
sudo is already the newest version (1.8.16-0ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
Adding group `codespace' (GID 1000) ...
Done.
/workspace /workspace
You have bazel 6.1.0 installed.
Found possible Python library paths:
 /usr/lib/python2.7/dist-packages
 /usr/local/lib/python2.7/dist-packages
Please input the desired Python library path to use. Default is [/usr/lib/python2.7/dist-packages]
Do you wish to build TensorFlow with ROCm support? [y/N]: No ROCm support will be enabled for TensorFlow.

Do you wish to build TensorFlow with CUDA support? [y/N]: No CUDA support will be enabled for TensorFlow.

Do you want to use Clang to build TensorFlow? [Y/n]: Clang will be used to compile TensorFlow.

Please specify the path to clang executable. [Default is /usr/bin/clang]: 

You have Clang 3.8.0-2ubuntu4 installed.

Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is -Wno-sign-compare]: 

Would you like to interactively configure ./WORKSPACE for Android builds? [y/N]: Not configuring the WORKSPACE for Android builds.

Preconfigured Bazel build configs. You can use any of the below by adding "--config=<>" to your build command. See .bazelrc for more details.
    --config=mkl      # Build with MKL support.
    --config=mkl_aarch64  # Build with oneDNN and Compute Library for the Arm Architecture (ACL).
    --config=monolithic   # Config for mostly static monolithic build.
    --config=numa      # Build with NUMA support.
    --config=dynamic_kernels    # (Experimental) Build kernels into separate shared objects.
    --config=v1       # Build with TensorFlow 1 API instead of TF 2 API.
Preconfigured Bazel build configs to DISABLE default on features:
    --config=nogcp     # Disable GCP support.
    --config=nonccl     # Disable NVIDIA NCCL support.
/workspace
TF_BUILD_INFO = {container_type: "cpu", command: "bazel test //tensorflow/...", source_HEAD: "483ba60a5afa9e3c77ffc76a3d6eceadfd328bf2", source_remote_origin: "https://github.com/giuliocn/tensorflow", OS: "Linux", kernel: "6.2.0-1018-azure", architecture: "x86_64", processor: "AMD EPYC 7763 64-Core Processor", processor_count: "2", memory_total: "8120288 kB", swap_total: "0 kB", Bazel_version: "Build label: 6.1.0", Java_version: "1.8.0_292", Python_version: "2.7.12", gpp_version: "g++ (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609", swig_version: "", NVIDIA_driver_version: "", CUDA_device_count: "0", CUDA_device_names: "", CUDA_toolkit_version: ""}
INFO: Options provided by the client:
 Inherited 'common' options: --isatty=0 --terminal_columns=80
INFO: Reading rc options for 'test' from /etc/bazel.bazelrc:
 Inherited 'common' options: --color=yes
INFO: Reading rc options for 'test' from /workspace/.bazelrc:
 Inherited 'common' options: --experimental_repo_remote_exec
INFO: Reading rc options for 'test' from /etc/bazel.bazelrc:
 Inherited 'build' options: --verbose_failures --spawn_strategy=standalone --strategy=Genrule=standalone
INFO: Reading rc options for 'test' from /workspace/.bazelrc:
 Inherited 'build' options: --define framework_shared_object=true --define tsl_protobuf_header_only=true --define=use_fast_cpp_protos=true --define=allow_oversize_protos=true --spawn_strategy=standalone -c opt --announce_rc --define=grpc_no_ares=true --noincompatible_remove_legacy_whole_archive --features=-force_no_whole_archive --enable_platform_specific_config --define=with_xla_support=true --config=short_logs --config=v2 --define=no_aws_support=true --define=no_hdfs_support=true --experimental_cc_shared_library --experimental_link_static_libraries_once=false --incompatible_enforce_config_setting_visibility
INFO: Reading rc options for 'test' from /workspace/.tf_configure.bazelrc:
 Inherited 'build' options: --host_force_python=PY2 --action_env PYTHON_BIN_PATH=/usr/bin/python --action_env PYTHON_LIB_PATH=/usr/lib/python2.7/dist-packages --python_path=/usr/bin/python --action_env CLANG_COMPILER_PATH=/usr/lib/llvm-3.8/bin/clang --repo_env=CC=/usr/lib/llvm-3.8/bin/clang --repo_env=BAZEL_COMPILER=/usr/lib/llvm-3.8/bin/clang
INFO: Reading rc options for 'test' from /etc/bazel.bazelrc:
 'test' options: --spawn_strategy=standalone --verbose_failures --test_output=errors --test_verbose_timeout_warnings
INFO: Reading rc options for 'test' from /workspace/.tf_configure.bazelrc:
 'test' options: --test_size_filters=small,medium
INFO: Found applicable config definition build:short_logs in file /workspace/.bazelrc: --output_filter=DONT_MATCH_ANYTHING
INFO: Found applicable config definition build:v2 in file /workspace/.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Found applicable config definition test:v2 in file /workspace/.tf_configure.bazelrc: --test_tag_filters=-benchmark-test,-no_oss,-oss_excluded,-gpu,-oss_serial,-v1only --build_tag_filters=-benchmark-test,-no_oss,-oss_excluded,-gpu,-v1only
INFO: Found applicable config definition build:linux in file /workspace/.bazelrc: --host_copt=-w --copt=-Wno-all --copt=-Wno-extra --copt=-Wno-deprecated --copt=-Wno-deprecated-declarations --copt=-Wno-ignored-attributes --copt=-Wno-array-bounds --copt=-Wunused-result --copt=-Werror=unused-result --copt=-Wswitch --copt=-Werror=switch --copt=-Wno-error=unused-but-set-variable --define=PREFIX=/usr --define=LIBDIR=$(PREFIX)/lib --define=INCLUDEDIR=$(PREFIX)/include --define=PROTOBUF_INCLUDE_PATH=$(PREFIX)/include --cxxopt=-std=c++17 --host_cxxopt=-std=c++17 --config=dynamic_kernels --experimental_guard_against_concurrent_changes
INFO: Found applicable config definition build:dynamic_kernels in file /workspace/.bazelrc: --define=dynamic_loaded_kernels=true --copt=-DAUTOLOAD_DYNAMIC_KERNELS
Loading: 
Loading: 
Loading: 
Loading: 0 packages loaded
Loading: 14 packages loaded
  currently loading: tensorflow/compiler/mlir/tfrt/tests/tfrt_fallback ... (2 packages)
Loading: 33 packages loaded
  currently loading: tensorflow/compiler/mlir/tfrt/ir/mlrt ... (3 packages)
Loading: 58 packages loaded
  currently loading: tensorflow/compiler/mlir/tfrt/ir/mlrt ... (10 packages)
Loading: 152 packages loaded
  currently loading: tensorflow/security/fuzzing/cc/core/function
Loading: 212 packages loaded
  currently loading: tensorflow/core/kernels ... (2 packages)
Loading: 261 packages loaded
  currently loading: tensorflow/core/kernels ... (2 packages)
Loading: 351 packages loaded
  currently loading: tensorflow/core/transforms/toposort
Loading: 484 packages loaded
  currently loading: tensorflow/python/ops ... (2 packages)
Loading: 615 packages loaded
  currently loading: tensorflow/lite/testing ... (2 packages)
Loading: 654 packages loaded
  currently loading: tensorflow/lite/testing ... (2 packages)
Loading: 743 packages loaded
  currently loading: tensorflow/lite/acceleration/configuration ... (2 packages)
WARNING: /workspace/tensorflow/compiler/mlir/lite/stablehlo/BUILD:581:13: target '//tensorflow/compiler/mlir/lite/stablehlo:odml_to_stablehlo' is deprecated: odml_to_stablehlo is being deprecated, please use TFlite converter with flag: converter.target_spec.supported_ops = [tf.lite.OpsSet.EXPERIMENTAL_STABLEHLO_OPS]
WARNING: /workspace/tensorflow/core/kernels/BUILD:973:18: target '//tensorflow/core/kernels:bitcast_op' is deprecated: use //third_party/tensorflow/c/kernels:bitcast_op instead
WARNING: /workspace/tensorflow/lite/core/async/testing/BUILD:27:23: target '//tensorflow/lite/core/async/testing:mock_async_kernel' is deprecated: Use //tensorflow/lite/async/testing:mock_async_kernel instead.
WARNING: /workspace/tensorflow/lite/core/async/BUILD:87:23: target '//tensorflow/lite/core/async:backend_async_kernel_interface' is deprecated: Use //tensorflow/lite/async:backend_async_kernel_interface instead.
WARNING: /workspace/tensorflow/python/eager/BUILD:696:18: target '//tensorflow/python/eager:framework_for_generated_wrappers' is deprecated: Depending on this target can cause build dependency cycles. Depend on the fine-grained sub-targets instead.
WARNING: /workspace/tensorflow/python/ops/distributions/BUILD:9:18: target '//tensorflow/python/ops/distributions:distributions' is deprecated: TensorFlow Distributions has migrated to TensorFlow Probability (https://github.com/tensorflow/probability). Deprecated copies remaining in tf.distributions will not receive new features, and will be removed by early 2019. You should update all usage of `tf.distributions` to `tfp.distributions`.
Analyzing: 19154 targets (797 packages loaded)
Analyzing: 19154 targets (798 packages loaded, 0 targets configured)
Analyzing: 19154 targets (814 packages loaded, 79 targets configured)
Analyzing: 19154 targets (816 packages loaded, 393 targets configured)
Analyzing: 19154 targets (817 packages loaded, 4468 targets configured)
Analyzing: 19154 targets (820 packages loaded, 9559 targets configured)
Analyzing: 19154 targets (823 packages loaded, 12199 targets configured)
ERROR: /workspace/tensorflow/tools/toolchains/win_1803/py38/BUILD:9:11: in interpreter_path attribute of py_runtime rule //tensorflow/tools/toolchains/win_1803/py38:py3_runtime: must be an absolute path.
ERROR: /workspace/tensorflow/tools/toolchains/win_1803/py38/BUILD:9:11: Analysis of target '//tensorflow/tools/toolchains/win_1803/py38:py3_runtime' failed
INFO: Repository tflite_mobilenet_float instantiated at:
 /workspace/WORKSPACE:84:14: in <toplevel>
 /workspace/tensorflow/workspace2.bzl:936:21: in workspace
 /workspace/tensorflow/workspace2.bzl:645:20: in _tf_repositories
 /workspace/third_party/repo.bzl:136:21: in tf_http_archive
Repository rule _tf_http_archive defined at:
 /workspace/third_party/repo.bzl:89:35: in <toplevel>
INFO: Repository go_sdk instantiated at:
 /workspace/WORKSPACE:92:14: in <toplevel>
 /workspace/tensorflow/workspace0.bzl:135:20: in workspace
 /workspaces/tensorflow/bazel-ci_build-cache/.cache/bazel/_bazel_codespace/eab0d61a99b6696edb3d2aff87b585e8/external/com_github_grpc_grpc/bazel/grpc_extra_deps.bzl:36:27: in grpc_extra_deps
 /workspaces/tensorflow/bazel-ci_build-cache/.cache/bazel/_bazel_codespace/eab0d61a99b6696edb3d2aff87b585e8/external/io_bazel_rules_go/go/private/sdk.bzl:431:28: in go_register_toolchains
 /workspaces/tensorflow/bazel-ci_build-cache/.cache/bazel/_bazel_codespace/eab0d61a99b6696edb3d2aff87b585e8/external/io_bazel_rules_go/go/private/sdk.bzl:130:21: in go_download_sdk
Repository rule _go_download_sdk defined at:
 /workspaces/tensorflow/bazel-ci_build-cache/.cache/bazel/_bazel_codespace/eab0d61a99b6696edb3d2aff87b585e8/external/io_bazel_rules_go/go/private/sdk.bzl:117:35: in <toplevel>
ERROR: Analysis of target '//tensorflow/tools/toolchains/win_1803/py38:py3_runtime' failed; build aborted: 
INFO: Elapsed time: 23.466s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (823 packages loaded, 12233 targets configured)
ERROR: Couldn't start the build. Unable to run tests
</details>

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @giuliocn This PR is in draft, any update on this? Please. Thank you!

**Comment by gbaned** [comment]
Hi @angerson Can you please assist on above[ comments](https://github.com/tensorflow/tensorflow/pull/62898#issuecomment-1984228895) from @giuliocn. Thank you!

**Comment by giuliocn** [comment]
> Hi @angerson Can you please assist on above[ comments](https://github.com/tensorflow/tensorflow/pull/62898#issuecomment-1984228895) from @giuliocn. Thank you!

The build failure is related to an invalid path configuration for the Python interpreter on the Windows 1803 system with Python 3.8 (denoted as `win_1803/py38`).

@gbaned Here's a breakdown of the error message generated by **Gemini**:

* **Error Message:**  
  ```
  ERROR: /workspace/tensorflow/tools/toolchains/win_1803/py38/BUILD:9:11: in interpreter_path attribute of py_runtime rule //tensorflow/tools/toolchains/win_1803/py38:py3_runtime: must be an absolute path.
  ```
* **Explanation:** The error points to a specific line (line 9, column 11) in a build file (`BUILD`). This line defines a rule named `py_runtime` which specifies the path to the Python interpreter used during the build process. The error message states that the provided path in the `interpreter_path` attribute is not a valid absolute path. An absolute path starts with the drive letter (e.g., `C:`) or a network location and specifies the complete directory structure to reach the desired file.

**Possible Causes:**

* **Incorrect Path:** The path configured in the `interpreter_path` attribute might be incorrect or relative. It should be the complete path to the Python 3.8 executable on the Windows system (e.g., `C:\Python38\python.exe`).
* **Missing Configuration:** The build configuration might not be set up to automatically locate the Python interpreter. You might need to explicitly specify the path during the build process.

**Resolving the Issue:**

1. **Locate Python Executable:** Find the absolute path to the Python 3.8 executable on your Windows 1803 system. It's typically installed in a directory like `C:\Python38` or `C:\Users\<username>\AppData\Local\Programs\Python\Python38`.
2. **Update Build Configuration:** Modify the `BUILD` file (`/workspace/tensorflow/tools/toolchains/win_1803/py38/BUILD`) and update the `interpreter_path` attribute with the absolute path you located in step 1.
3. **Rerun Build:** After updating the path, rerun the build command (`bazel test //tensorflow/...`) to attempt building TensorFlow again.

By providing a valid absolute path to the Python interpreter, the build process should be able to locate the correct executable and proceed successfully.
